vnext/Microsoft.ReactNative.SharedManaged/AttributedViewManager.cs

```diff
@@ -0,0 +1,146 @@
+
+namespace Microsoft.ReactNative.Managed
+{
+  internal abstract class AttributedViewManager<T> : IViewManager where T : FrameworkElement, new()
+  {
+    public virtual string Name => typeof(T).Name;
+
+    public virtual FrameworkElement CreateView() => new T();
+
+    public IReadOnlyDictionary<string, object> Commands => _commands ?? (_commands = GetConstantsByAttribute<ViewManagerCommandAttribute>());
+    private IReadOnlyDictionary<string, object> _commands;
+
+    public IReadOnlyDictionary<string, object> ExportedViewConstants => _exportedViewConstants ?? (_exportedViewConstants = GetConstantsByAttribute<ViewManagerExportedViewConstantAttribute>());
+    private IReadOnlyDictionary<string, object> _exportedViewConstants;
+
+    public IReadOnlyDictionary<string, ViewManagerPropertyType> NativeProps
+    {
+      get
+      {
+        if (null == _nativeProps)
+        {
+          var nativeProps = new Dictionary<string, ViewManagerPropertyType>();
+          foreach (var kvp in PropertySetters)
+          {
+            nativeProps.Add(kvp.Key, kvp.Value.Type);
+          }
+          _nativeProps = nativeProps;
+        }
+        return _nativeProps;
+      }
+    }
+    private IReadOnlyDictionary<string, ViewManagerPropertyType> _nativeProps;
+
+    internal Dictionary<string, ViewManagerPropertySetter<T>> PropertySetters
+    {
+      get
+      {
+        if (null == _propertySetters)
+        {
+          _propertySetters = new Dictionary<string, ViewManagerPropertySetter<T>>();
+
+          foreach (var methodInfo in GetType().GetTypeInfo().DeclaredMethods)
+          {
+            var setterAttribute = methodInfo.GetCustomAttribute<ViewManagerPropertySetterAttribute>();
+            if (null != setterAttribute)
+            {
+              var setter = new ViewManagerPropertySetter<T>();
+              setter.Name = setterAttribute.Name;
+              setter.Type = setterAttribute.Type ?? TypeToViewManagerPropertyType(methodInfo.GetParameters()[1].ParameterType);
+              setter.Method = (view, propertyValue) =>
+              {
+                methodInfo.Invoke(this, new object[] { view, propertyValue });
+              };
+              _propertySetters.Add(setter.Name, setter);
+            }
+          }
+        }
+
+        return _propertySetters;
+      }
+    }
+    private Dictionary<string, ViewManagerPropertySetter<T>> _propertySetters;
+
+    public void UpdateProperties(FrameworkElement view, IReadOnlyDictionary<string, object> propertyMap)
+    {
+      if (view is T viewAsT)
+      {
+        foreach (var property in propertyMap)
+        {
+          if (PropertySetters.TryGetValue(property.Key, out ViewManagerPropertySetter<T> setter))
+          {
+            setter.Method(viewAsT, property.Value);
+          }
+        }
+      }
+    }
+
+    private static ViewManagerPropertyType TypeToViewManagerPropertyType(Type t)
+    {
+      if (t == typeof(bool) || t == typeof(bool?))
+      {
```

Comment:
this and the below wouldn't catch types that have conversion operators
instead consider using [Type.IsAssignableFrom](https://docs.microsoft.com/en-us/dotnet/api/system.type.isassignablefrom?view=netframework-4.8)

Reply:
not available in UWP :(